Added Set of Test Attestation Certificates

emargolis opened the pull request:
These test certificates can be used in testing device commissioning process.

Also added shell script that was used to generates those certificates.

As next step I plan to add test Certification Declaration (CD) and test Operational Certs to:
credentials/test/cd
credentials/test/operational

#### Testing

bzbarsky-apple commented on credentials/test/gen-test-attestation-certs.sh:
```diff
@@ -0,0 +1,82 @@
+
+mkdir -p "$dest_dir"
+
+chip_cert_tool="$CHIP_ROOT/out/debug/linux_x64_gcc/obj/src/tools/chip-cert/bin/chip-cert"
```
This doesn't work on Mac, obviously, or Windows.

It also relies on some sort of build step (which is not documented anywhere here) being done to create that binary.

What we should do is document what that build step is and then use something like `find $CHIP_ROOT -path "*/bin/chip-cert"` or `find $CHIP_ROOT -path "*gcc*bin/chip-cert"` or something like that, depending on what the build step is, to find the relevant binaries.

mspang replied:
Please don't use find for this as it could easily find the wrong (e.g. stale) binary. Recommend passing it as an argument.

emargolis replied:
Updated to pass `chip-cert` tool binary as an input argument.

tcarmelveilleux commented on credentials/test/gen-test-attestation-certs.sh:
```diff
@@ -0,0 +1,84 @@
+cert_lifetime=7305
+
+paa_vids=(FFF1 FFF2)
+pai_pids=(ABC0 ABC1)
```
Suggest using 0000 and 0001 for these, or 0x8000, 0x8001

emargolis replied:
done

tcarmelveilleux commented on credentials/test/gen-test-attestation-certs.sh:
```diff
@@ -0,0 +1,84 @@
+dac_ids=(0001 0002 0003 0004 0005 0006 0007 0008 0009 000A 000B 000C 000D 000E 000F)
+
+for vid in "${paa_vids[@]}"; do
+    paa_key_file="$dest_dir/Chip-Test-PAA-$vid-Key"
+    paa_cert_file="$dest_dir/Chip-Test-PAA-$vid-Cert"
```
What would be best is to generate the test PAAs only once, and include the gen code in comments in here. Then we could start "officially" storing those dev PAAs and including them in DCL soon. For the PAI and DAC, it's OK to generate them.

I think overall we must avoid these test PAAs ever changing, so as to avoid having to rotate them and deal with possible commissioning tests downtime in the future.

Similarly the *dev* PAAs here should have infinite validity period.

emargolis replied:
This script is not a tool but just a reference example of how these attestation certificates were generated. The intention here is that this script shouldn't be used as is to re-generate certificates that already present. I think it is OK to leave it as is and for any new certificates that should be added this script will be manually updated according to the requirements.

After more detailed discussions with DMTT it was agreed that PAA, PAI and DAC certificates shouldn't have expiration. I updated all test certs in this PR to have "No Well Defined Expiration" value.

emargolis commented:
In addition to the previous update I've added:
- Support of "No Well Defined Expiration" option to the `chip-cert` tool 
- Updated all PAA, PAI, and DAC certs to have no expiration
